strawberry_django/auth/mutations.py

```diff
@@ -0,0 +1,26 @@
+import strawberry
+from django.contrib import auth
+from ..resolvers import django_resolver
+
+def login():
+    @strawberry.mutation
+    @django_resolver
+    def login(info, username: str, password: str):
+        request = info.context.request
+        user = auth.authenticate(request, username=username, password=password)
+        if user is not None:
+            auth.login(request, user)
+            return user
+        auth.logout(request)
```

Comment:
Why do you wrap the login function in a login function?

Shouldnt this login fn have a return type?

Reply:
We want to return mutation field, not function. Would following code be better?

```python
@django_resolver
def resolve_login(info, username: str, password: str):
    ...
    return request.user

@django_resolver
def resolve_logout(info) -> bool:
    ...
    return True

def login():
    return strawberry.mutation(resolver=resolve_login)

def logout():
    return strawberry.mutation(resolver=resolve_logout)
```

Return type of login should come from app, in that way developer can define own type for user like this

```python
@strawberry.type
class Mutation:
    login: MyUserType: strawberry_django.auth.login()
```

Reply:
Wouldn't a decorator work here?
```py
@strawberry.mutation
@django_resolver
def login(info, username: str, password: str):
    ...
    return request.user
    
# in schema:
      login = login
```

Reply:
Yes that would work but then the syntax is a bit different compared to field. Strawberry field and mutation are both functions.

```python
@strawberry.type
class Mutation:
    login: User = auth.login
    # vs
    login: User = auth.login()
    # vs
    login: User = strawberry.mutation(resolver=login_resolver)
```

Function would allow us to add new parameters in the future if we need to make login mutation configurable. See following example.

```python
@strawberry.type
class Mutation:
    login: User = auth.login(only_superusers=True)
```

Do you agree?

Reply:
To be honest, I don't really care much for this login function, especially if we have to cover a lot of different cases. I'd prefer to just add it in the documentation in a "how to", then everyone can alter it to their use case. Its extremely simple if you need to cover 1 case. It will be a mess if it needs to cover all cases.